### PR TITLE
Block Editor: Correctly mark Block Comment SlotFills private

### PR DESCRIPTION
**Pull request by Mamaduka**
## What?
PR updates block comment SlotFills and correctly mark them as private, preventing "leaking" them to consumers using string names.

I've also removed the `__unstable` prefix; it's not needed for private APIs.

Remembered this case while reviewing #67238.

## Testing Instructions
1. Enable "Enable multi-user commenting on blocks" experiment.
2. Confirm you can add comments from the block options dropdown.
3. Opens the comment sidebar from the block toolbar.
4. Confirm that custom fills can't be injected into private slots. See the snippet below.

<details><summary>Test the "leak"</summary>
<p>

```javascript
const {createElement: el} = wp.element;
const registerPlugin = wp.plugins.registerPlugin;
const {Fill, ToolbarButton} = wp.components;

function TestSlotFillLeak() {
    return el(Fill, {
        name: '__unstableCommentIconToolbarFill',
    }, el(ToolbarButton, {
        icon: 'carrot',
        label: 'Test',
        onClick: () => console.log('test')
    }, ));
}

registerPlugin('test-slot-fill-leak', {
    render: TestSlotFillLeak,
});
``` 

</p>
</details> 

### Testing Instructions for Keyboard
Same.

## Screenshots or screencast 
CleanShot 2024-11-25 at 15 26 19

### CODE REVIEW

File: packages/block-editor/src/components/collab/block-comment-icon-slot.js [hunk 1]
```diff
@@ -3,10 +3,10 @@
...
 import { createSlotFill } from '@wordpress/components';
 
-const { Fill: __unstableCommentIconFill, Slot } = createSlotFill(
-	'__unstableCommentIconFill'
+const { Fill: CommentIconFill, Slot } = createSlotFill(
+	Symbol( 'CommentIconFill' )
 );
 
-__unstableCommentIconFill.Slot = Slot;
+CommentIconFill.Slot = Slot;
```
Comment (jsnajdr) [inline packages/block-editor/src/components/collab/block-comment-icon-slot.js]: If this is a private API, let's fix it by exporting the return value of `createSlotFill` without any further modifications. We can call it `CommentIconSlotFill` and use its `.Slot` and `.Fill` fields. No needs to have a `Fill` compoennt with a `Fill.Slot` field.

Same thing can be done for `CommentIconToolbarSlotFill`.

Comment (Mamaduka) [reply packages/block-editor/src/components/collab/block-comment-icon-slot.js]: Updated. I'm sure the names will change again before the feature is stabilized, but it should be good for now.

**Review by jsnajdr — APPROVED** [review]
Looks good, thanks for tidying this up :+1:

**Comment by Mamaduka** [comment]
Thanks, @jsnajdr!

**Pull request merged**